marshmallow_sqlalchemy/convert.py

```diff
@@ -28,44 +40,19 @@ class ModelConverter(object):
     """
 
     SQLA_TYPE_MAPPING = {
-        sa.String: fields.String,
-        sa.Unicode: fields.String,
-        sa.Boolean: fields.Boolean,
-        sa.Unicode: fields.String,
-        sa.Binary: fields.String,
         sa.Enum: fields.Field,
-        sa.Numeric: fields.Decimal,
-        sa.Float: fields.Decimal,
-        sa.Date: fields.Date,
 
+        postgresql.BIT: fields.Integer,
         postgresql.UUID: fields.UUID,
         postgresql.MACADDR: fields.String,
         postgresql.INET: fields.String,
+        postgresql.JSON: fields.Raw,
+        postgresql.JSONB: fields.Raw,
+        postgresql.HSTORE: fields.Raw,
+        postgresql.ARRAY: _postgres_array_factory,
 
         mysql.BIT: fields.Integer,
-        mysql.TINYINT: fields.Integer,
```

Comment:
I probably should've put this into its own PR, but the mappings that I deleted aren't necessary. Because `mysql.TINYINT().python_type` is `int`, the mappings in `Schema.TYPE_MAPPING` already handle this case. I think mappings are only needed here for columns that either don't define `python_type` or that need special logic, like `postgres.UUID` or `postgres.ARRAY`.

Semi-documented in commit message at https://github.com/jmcarp/marshmallow-sqlalchemy/commit/67b2eb7cc94edc35d2ad4d74481c17cb7011c3d6.